src/headless/tasks/inasafe_wrapper.py

```diff
@@ -176,18 +222,34 @@ def generate_report():
 
 
 @app.task(queue='inasafe-headless')
-def generate_contour(layer_uri, output_uri=None):
+def generate_contour(layer_uri):
     """Create contour from raster layer_uri to output_uri
 
     :param layer_uri: The shakemap raster layer uri.
     :type layer_uri: basestring
 
-    :param output_uri: The contour output layer uri
-    :type output_uri: basestring
-
     :returns: The output layer uri if success
     :rtype: basestring
+
+    It will put the contour layer to
+    contour_[input_file_name]_[current_datetime]/[input_file_name].shp
+
+    current_datetime format: 25January2018_09h25-17.597909
     """
+    # Always create directory
+    input_file_name = os.path.basename(layer_uri)
+    input_base_name = os.path.splitext(input_file_name)[0]
+    # Make it same format as analysis directory
+    current_datetime = datetime.now().strftime('%d%B%Y_%Hh%M-%S.%f')
+    output_directory = 'contour_%s_%s' % (input_base_name, current_datetime)
+    output_file_name = input_base_name + '.shp'
+    output_directory_path = os.path.join(
+        setting('defaultUserDirectory'), output_directory)
+    # Make sure the output directory exists
+    if not os.path.exists(output_directory_path):
```

Comment:
I mean, it is somewhat important for these types of software because it is going to be scaled, and multiple docker services will work on the same directory. However in this particular case, it will be very unlikely to have the same directory.

Reply:
Ok, thanks for the insight. I never thought there will be a race condition. But, it's a good habit. I will change it.